### ISSUE TITLE
FN unnecessary_unwrap

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6540]
Lint name : unnecessary_unwrap

I tried this code:

```rust
pub struct S {
    inner: Option<i32>,
}

impl S {
    pub fn fun(&self) -> i32 {
        if self.inner.is_some() {
            return self.inner.unwrap();
        }
        4
    }
}
```

I expect this to trigger `clippy::unnecessary_unwrap` and suggest something like
````rust
if let Some(val) = self.inner {
    return val;
}
4
````

but the lint did not trigger :frowning:

### Meta
`clippy 0.0.212 (fde6927 2021-01-02)`

**Comment by matthiaskrgr** [comment]
Apparently this works if I have a variable `x` which is an `Option` but does not work if I have a struct field `self.inner` :/
